python/lsst/summit/testing/analysis/m1m3/inertia_compensation_system.py

```diff
@@ -0,0 +1,464 @@
+        event_dict = vars(self.event)
+        event_dict = {
+            key: val
+            for key, val in event_dict.items()
+            if key in ["dayObs", "seqNum", "version"]
+        }
+
+        # Create a pandas Series from the dictionary
+        event_series = pd.Series(event_dict)
+
+        # Concatenate the two Series
+        result_series = pd.concat([event_series, result_series])
+
+        # Display the resulting Series
+        return result_series
+
+
+def find_adjacent_true_regions(
+    series: pd.Series, min_adjacent: None | int = None
+) -> list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]:
+    """
+    Find regions in a boolean Series containing adjacent True values.
+
+    Parameters
+    ----------
+    series : pd.Series
+        The boolean Series to search for regions.
+
+    min_adjacent : int, optional
+        Minimum number of adjacent True values in a region.
+        Defaults to half size of the series.
+
+    Returns
+    -------
+    list[tuple[pd.DatetimeIndex, pd.DatetimeIndex]]
+        A list of tuples representing the start and end indices of regions
+        containing more than or equal to min_adjacent adjacent True values.
+    """
+    min_adjacent = min_adjacent if min_adjacent else 0.5 * series.size
+    regions = []
+    for key, group in series.groupby((series != series.shift()).cumsum()):
+        if key and len(group) >= min_adjacent:
+            region_indices = group.index
+            regions.append((region_indices.min(), region_indices.max()))
+    return regions
+
+
+def get_tma_slew_event(day_obs: int, seq_number: int) -> TMAEvent:
+    """
+    Retrieve Telescope Mount Assembly (TMA) slew events within a specified time
+    range.
+
+    Parameters
+    ----------
+    dayObs : int
+        Observation day in the YYYYMMDD format.
+    seqNum : int
+        Sequence number associated with the slew event.
+
+    Returns
+    -------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew events that occurred within the specified time range.
+
+    Notes
+    -----
+    This function retrieves TMA slew events occurring between the specified
+    start and end times. It uses the TMAEventMaker class to obtain events for
+    the specified day of observation (dayObs). The events are filtered to
+    include only those that start after 1 second before the specified start
+    time and end before 1 second after the specified end time.
+
+    Returs
+    ------
+    lsst.summit.utils.tmaUtils.TMAEvent
+        A TMA slew event that occurred within the specified time range.
+
+    Raises
+    ------
+    ValueError
+        If no events are found for the specified time range.
+    ValueError
+        If more than one event is found for the specified time range.
+    """
+    logger.info(f"Query events in {day_obs}")
+    events = event_maker.getEvents(day_obs)
+
+    logger.info(f"Found {len(events)} events.")
+    single_event = [e for e in events if e.seqNum == seq_number]
```

Comment:
I can see why you've taken this approach, but honestly, they're always zero-based and contiguous integers and always will be, so there actually isn't really any reason not to just check if the list is long enough, and if it is, return the element and raise otherwise. I don't hate this approach, but the other way should be just as safe, a little clearer, and easier to maintain. You could even do an `assert event.seq_num == seq_num` before returning as a safety-check.

Reply:
Also, maybe you just want that to be part of the event maker API actually. Feel free to just ask for things like this 🙂

Reply:
Not sure I understand how to implement your other approach. However, I am applying the safety-check you suggested. 

Regarding having it in the API, I believe it is a functionality that people could be happy with. As long as it does not take much of your time.

Reply:
Sure, it'll be trivial for me to add, that's fine. For what I meant though, everything in here could just be this (untested, just typing here, but you'll get the point)

```
if seq_number <= len(events):
    return events[seq_number]
else:
    raise ValueError(f"{seq_number=} is out of range - {day_obs=} only has {len(events)} events")
    ```

Reply:
Oh, one new comment though actually: I'd really recommend trying to stick to conventional naming schemes, so I'd call this `seq_num` not `seq_number` everywhere, otherwise people will slowly go crazy not know where a particular thing is called what 🙂

Reply:
I've filed to https://jira.lsstcorp.org/browse/DM-40948 to make this a call to the event make itself. I should get that done tomorrow, it's super easy but I have covid so am going to sleep now rather than write it 🙃